extensions/amp-bind/0.1/bind-impl.js

```diff
@@ -709,16 +709,31 @@ export class Bind {
         break;
 
       default:
+        // Some input elements treat some of their attributes as initial values.
+        // Once the user interacts with these elements, the JS properties
+        // underlying these attributes must be updated for the change to be
+        // visible to the user.
+        const requiresJsUpdate =
+            element.tagName == 'INPUT' && property in element;
+        // Property value *must* be read before the attribute is changed.
+        // Before user interaction, attribute updates affect the property.
+        const oldJsPropertyValue = requiresJsUpdate && element[property];
         const oldValue = element.getAttribute(property);
 
         let attributeChanged = false;
         if (typeof newValue === 'boolean') {
-          if (newValue && oldValue !== '') {
+          if (newValue) {
             element.setAttribute(property, '');
-            attributeChanged = true;
-          } else if (!newValue && oldValue !== null) {
+          } else {
             element.removeAttribute(property);
-            attributeChanged = true;
+          }
+          if (requiresJsUpdate) {
+            element[property] = newValue;
+            attributeChanged = oldJsPropertyValue !== newValue;
+          } else {
+            attributeChanged =
+                (oldValue === null && newValue) ||
+                (oldValue === '' && !newValue);
```

Comment:
I think it's more readable if you reverted the `attributeChanged = true` lines above and removed this else block.